tests/functional/fog/tests.rb

```diff
@@ -0,0 +1,163 @@
+require "fog"
+require "digest/md5"
+require "json"
+require "securerandom"
+require "excon"
+
+config = File.read("../config.json")
+rubyConfig = JSON.parse(config)
+
+transport = rubyConfig["transport"];
+ipAddress = rubyConfig["ipAddress"];
+endpoint = "#{transport}://#{ipAddress}:8000";
+
+if(rubyConfig["certPath"])
+    Excon.defaults[:ssl_ca_path] = rubyConfig["caCertPath"]
+end
+
+# Disabling to allow for end to end tests in CI
+Excon.defaults[:ssl_verify_peer] = false
```

Comment:
Do we need this if we are going to use http only? If there is a plan for https, our self-signed certs should still work w/o having to disabling ssl verify peer